sdks/python/apache_beam/pipeline.py

```diff
@@ -504,10 +504,12 @@ def run(self, test_runner_api='AUTO'):
       if test_runner_api == 'AUTO':
         # Don't pay the cost of a round-trip if we're going to be going through
         # the FnApi anyway...
+        # The InteractiveRunner relies a constant pipeline reference, skip it.
         test_runner_api = (
             not self.runner.is_fnapi_compatible() and (
                 self.runner.__class__.__name__ != 'SwitchingDirectRunner' or
-                self._options.view_as(StandardOptions).streaming))
+                self._options.view_as(StandardOptions).streaming) and
+            self.runner.__class__.__name__ != 'InteractiveRunner')
```

Comment:
Is there a better way for the runner to indicate it relies on a constant pipeline reference?

Reply:
@TheNeuralBit what do you think? I was following the precedence of the SwitchingDirectRunner. The InteractiveRunner already does this check in its code in multiple places, so there wouldn't be a safety regression if this line is added.

Reply:
The SwitchingDirectRunner case is a little different since its checking for a specific case where that runner will use the Fn API. 

If there's no way to avoid the need for a constant pipeline reference I think this is ok (I think we're getting rid of this roundtrip soon anyway), but let's break this into separate booleans for clarity, something like:

```py
is_fnapi_compatible = self.runner.is_fnapi_compatible() or (
                # DirectRunner uses the Fn API for batch only
                self.runner.__class__.__name__ == 'SwitchingDirectRunner' and
                not self._options.view_as(StandardOptions).streaming)

test_runner_api = (not is_fn_api_compatible or 
                   self.runner.__class__.__name__ == 'InteractiveRunner')
```

Reply:
Done.